scripts/apply-mail.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python3
+
+import email
+import argparse
+import re
+import tempfile
+import os
+
+argparser = argparse.ArgumentParser()
+argparser.add_argument('emails', metavar='FILE', nargs='+',
+                            help='Files with email message containing a patch')
+
+
+args = argparser.parse_args()
+
+bad_google = re.compile(r'".* via .*" <.*@googlegroups.com>')
+
+temps = []
+
+for email_file in args.emails:
+    e = email.message_from_file(open(email_file))
+    author = e.get('From')
+    m = re.fullmatch(bad_google, author)
+    if m:
+        e.replace_header('From', e.get('Reply-To'))
+    o = tempfile.NamedTemporaryFile()
```

Comment:
An idea (which sounds like a func exercise but isn't critical):

The risk of the NamedTemporaryFile() function is that interrupting this code can leave it behind. Traditional Unix code traps interrupts in this case to delete the temporary file, but you don't do it. TemporaryFile is safer in this regard - the file is immediately deleted.
But how can we use the unnamed temporary file in "git am"? You can, using /proc/self/fd/...

Reply:
Python will delete the file if interrupted, unless by a SIGKILL. I think the damage from a few text files in /tmp is low enough to ignore it.

Reply:
@avikivity it turns out you're right **theoretically**, because Python does catch SIGINT and turn it into an exception, and exceptions are handled correctly. It turns out that in practice, the implementation isn't perfect (https://github.com/python/cpython/issues/78338), but you're right that for what we're doing (a script for developers), it's good enough.